app/src/components/searchWidget/budgetRangeSlider.tsx

```diff
@@ -40,55 +34,38 @@ class BudgetRangeSlider extends React.Component<SliderProps, SliderState> {
     }
 
     onAfterChangeSlider(sliderValue: number) {
-        this.props.onChange && this.props.onChange([this.props.min, sliderValue]);
-    }
-
-    getStep() {
-        let value = this.state.currentValue;
-        switch (true) {
-            case value <= 500:
-                return process.env.REACT_APP_SLIDER_LOWEST_STEP
-                    ? parseInt(process.env.REACT_APP_SLIDER_LOWEST_STEP)
-                    : this.props.step;
-            case value > 500 && value <= 1000:
-                return process.env.REACT_APP_SLIDER_MEDIUM_STEP
-                    ? parseInt(process.env.REACT_APP_SLIDER_MEDIUM_STEP)
-                    : this.props.step;
-            case value > 1000:
-                return process.env.REACT_APP_SLIDER_HIGHEST_STEP
-                    ? parseInt(process.env.REACT_APP_SLIDER_HIGHEST_STEP)
-                    : this.props.step;
-            default:
-                return this.props.step;
-        }
+        this.props.onChange && this.props.onChange([this.props.min, Math.round(Math.exp(sliderValue))]);
     }
 
     render() {
-        const { min, max, className } = this.props;
-        const { currentValue } = this.state;
+        const {min, max, className} = this.props;
+        const {currentValue} = this.state;
+
+        let curmin = Math.log(min) >= 0 ? Math.log(min) : 0;
+        let curmax = Math.log(max);
 
         const sliderClassName = classnames(className, {
-            'max-filtered': currentValue !== max
+            'max-filtered': currentValue !== curmax
         });
 
         const sliderLabel =
-            currentValue === max ? 'Any price' : `$ ${currentValue.toString().replace(/\d(?=(\d{3})+)/g, '$&,')}`;
+            currentValue === curmax ? 'Any price' : `$ ${Math.round(Math.exp(currentValue)).toString().replace(/\d(?=(\d{3})+)/g, '$&,')}`;
 
         return (
             <div className={'widget-row widget-row-fill'}>
                 <div className="icon-label wj-icon">
-                    <FontAwesomeIcon icon={faDollarSign} />
+                    <FontAwesomeIcon icon={faDollarSign}/>
                 </div>
                 <div id="one-handler-range-slider" className={sliderClassName}>
                     <div className="text-container middle-text">
                         <span className="to">{sliderLabel}</span>
                     </div>
                     <div className="slider-container middle-text">
                         <Slider
-                            min={min}
-                            max={max}
+                            min={curmin}
+                            max={curmax}
                             value={currentValue}
-                            step={this.getStep()}
+                            step={(curmax - curmin) / 5000}
```

Comment:
What is this 5000 suppose to mean?

Reply:
this number is used to decide the granularity of each step, it divided the whole range into 5000 pieces. The slider will be more sensitive when the number gets smaller, and vise versa.